src/service/EventUtilService.ts

```diff
@@ -8,14 +8,13 @@ export default {
    */
   debounce (func: Function, wait: number) {
     let timeout: number | null
-    return () => {
-      let context = this
-      let args = arguments
-      let later = () => {
+    return (...args: any[]) => {
+      const context = this
+      const later = () => {
         timeout = null
         func.apply(context, args)
       }
-      let callNow = !timeout
+      const callNow = !timeout
```

Comment:
this bool is only used here:

  if (callNow) {
        func.apply(context, args)
      }

placing the var before this would help, or just check if(!timout) ?

Reply:
ah but the value of timeout does change between here and there, I *think*?

Reply:
replaced with lodash.debounce